### ISSUE TITLE
Failing with current ClojureScript master

### ISSUE THREAD

**Issue by mfikes** [metosin/spec-tools#127]
If you run spec-tools tests using ClojureScript master (built as of https://github.com/clojure/clojurescript/commit/f13c08c90f27aa2a71694859f288e9b5f2407c73),
unit tests fail:

```
;; ======================================================================
;; Testing with Phantom:


Testing spec-tools.core-test

Testing spec-tools.impl-test

Testing spec-tools.transform-test

Testing spec-tools.data-spec-test

Testing spec-tools.json-schema-test

Testing spec-tools.visitor-all-test

FAIL in (CLJ-2152-test) (:)
this should fail when the issues is fixed
expected: (not (= (clojure.core/sequence (clojure.core/seq (clojure.core/concat (clojure.core/list (quote cljs.spec.alpha/&)) (clojure.core/list (quote cljs.core/int?))))) (s/form (s/& int?))))
  actual: (not (not true))

FAIL in (visit-all-test) (:)
visitor visits all specs but not s/& & s/keys* inner specs
expected: (= #{:spec-tools.visitor-all-test/nilable :spec-tools.visitor-all-test/p24 :spec-tools.visitor-all-test/coll-of :spec-tools.visitor-all-test/cat :spec-tools.visitor-all-test/p2 :spec-tools.visitor-all-test/p12 :spec-tools.visitor-all-test/p11 :spec-tools.visitor-all-test/every-kv :spec-tools.visitor-all-test/* :spec-tools.visitor-all-test/keys3 :spec-tools.visitor-all-test/p18 :spec-tools.visitor-all-test/p9 :spec-tools.visitor-all-test/tuple :spec-tools.visitor-all-test/p26 :spec-tools.visitor-all-test/p5 :spec-tools.visitor-all-test/merge :spec-tools.visitor-all-test/? :spec-tools.visitor-all-test/p3 :spec-tools.visitor-all-test/or :spec-tools.visitor-all-test/p22 :spec-tools.visitor-all-test/all :spec-tools.visitor-all-test/p4 :spec-tools.visitor-all-test/p8 :spec-tools.visitor-all-test/p1 :spec-tools.visitor-all-test/map-of :spec-tools.visitor-all-test/p19 :spec-tools.visitor-all-test/p15 :spec-tools.visitor-all-test/p13 :spec-tools.visitor-all-test/p27 :spec-tools.visitor-all-test/p10 :spec-tools.visitor-all-test/p17 :spec-tools.visitor-all-test/and :spec-tools.visitor-all-test/p16 :spec-tools.visitor-all-test/keys :spec-tools.visitor-all-test/keys2 :spec-tools.visitor-all-test/& :spec-tools.visitor-all-test/pred :spec-tools.visitor-all-test/p34 :spec-tools.visitor-all-test/alt :spec-tools.visitor-all-test/p21 :spec-tools.visitor-all-test/every :spec-tools.visitor-all-test/keys* :spec-tools.visitor-all-test/p23 :spec-tools.visitor-all-test/p7 :spec-tools.visitor-all-test/p6 :spec-tools.visitor-all-test/p20 :spec-tools.visitor-all-test/+ :spec-tools.visitor-all-test/p14} (->> (visitor/visit :spec-tools.visitor-all-test/all (visitor/spec-collector)) (keys) (set)))
  actual: (not (= #{:spec-tools.visitor-all-test/nilable :spec-tools.visitor-all-test/p24 :spec-tools.visitor-all-test/coll-of :spec-tools.visitor-all-test/cat :spec-tools.visitor-all-test/p2 :spec-tools.visitor-all-test/p12 :spec-tools.visitor-all-test/p11 :spec-tools.visitor-all-test/every-kv :spec-tools.visitor-all-test/* :spec-tools.visitor-all-test/keys3 :spec-tools.visitor-all-test/p18 :spec-tools.visitor-all-test/p9 :spec-tools.visitor-all-test/tuple :spec-tools.visitor-all-test/p26 :spec-tools.visitor-all-test/p5 :spec-tools.visitor-all-test/merge :spec-tools.visitor-all-test/? :spec-tools.visitor-all-test/p3 :spec-tools.visitor-all-test/or :spec-tools.visitor-all-test/p22 :spec-tools.visitor-all-test/all :spec-tools.visitor-all-test/p4 :spec-tools.visitor-all-test/p8 :spec-tools.visitor-all-test/p1 :spec-tools.visitor-all-test/map-of :spec-tools.visitor-all-test/p19 :spec-tools.visitor-all-test/p15 :spec-tools.visitor-all-test/p13 :spec-tools.visitor-all-test/p27 :spec-tools.visitor-all-test/p10 :spec-tools.visitor-all-test/p17 :spec-tools.visitor-all-test/and :spec-tools.visitor-all-test/p16 :spec-tools.visitor-all-test/keys :spec-tools.visitor-all-test/keys2 :spec-tools.visitor-all-test/& :spec-tools.visitor-all-test/pred :spec-tools.visitor-all-test/p34 :spec-tools.visitor-all-test/alt :spec-tools.visitor-all-test/p21 :spec-tools.visitor-all-test/every :spec-tools.visitor-all-test/keys* :spec-tools.visitor-all-test/p23 :spec-tools.visitor-all-test/p7 :spec-tools.visitor-all-test/p6 :spec-tools.visitor-all-test/p20 :spec-tools.visitor-all-test/+ :spec-tools.visitor-all-test/p14} #{:spec-tools.visitor-all-test/nilable :spec-tools.visitor-all-test/p24 :spec-tools.visitor-all-test/coll-of :spec-tools.visitor-all-test/cat :spec-tools.visitor-all-test/p2 :spec-tools.visitor-all-test/p25 :spec-tools.visitor-all-test/p12 :spec-tools.visitor-all-test/p11 :spec-tools.visitor-all-test/every-kv :spec-tools.visitor-all-test/* :spec-tools.visitor-all-test/keys3 :spec-tools.visitor-all-test/p18 :spec-tools.visitor-all-test/p9 :spec-tools.visitor-all-test/tuple :spec-tools.visitor-all-test/p26 :spec-tools.visitor-all-test/p5 :spec-tools.visitor-all-test/merge :spec-tools.visitor-all-test/? :spec-tools.visitor-all-test/p3 :spec-tools.visitor-all-test/or :spec-tools.visitor-all-test/p22 :spec-tools.visitor-all-test/all :spec-tools.visitor-all-test/p4 :spec-tools.visitor-all-test/p8 :spec-tools.visitor-all-test/p1 :spec-tools.visitor-all-test/map-of :spec-tools.visitor-all-test/p19 :spec-tools.visitor-all-test/p15 :spec-tools.visitor-all-test/p13 :spec-tools.visitor-all-test/p27 :spec-tools.visitor-all-test/p10 :spec-tools.visitor-all-test/p17 :spec-tools.visitor-all-test/and :spec-tools.visitor-all-test/p16 :spec-tools.visitor-all-test/keys :spec-tools.visitor-all-test/keys2 :spec-tools.visitor-all-test/& :spec-tools.visitor-all-test/pred :spec-tools.visitor-all-test/p34 :spec-tools.visitor-all-test/alt :spec-tools.visitor-all-test/p21 :spec-tools.visitor-all-test/every :spec-tools.visitor-all-test/keys* :spec-tools.visitor-all-test/p23 :spec-tools.visitor-all-test/p7 :spec-tools.visitor-all-test/p6 :spec-tools.visitor-all-test/p20 :spec-tools.visitor-all-test/+ :spec-tools.visitor-all-test/p14}))

Testing spec-tools.visitor-test

Testing spec-tools.swagger.core-test

Testing spec-tools.swagger.spec-test

Ran 63 tests containing 354 assertions.
2 failures, 0 errors.
```

**Comment by mfikes** [comment]
Git bisect shows spec tests started failing with https://dev.clojure.org/jira/browse/CLJS-2839

**Comment by miikka** [comment]
Thanks for the report! This is good news, because it means that you have fixed a bug in ClojureScript spec implementation.

`CLJ-2152-test` is failing because it was checking for the existence a spec bug, which has been fixed on the Clojure side last year, but wasn't fixed in ClojureScript until now. 

For `visit-all-test`, the only difference is that now there is unexpected `::p25`. It is *actually* expected to be there, but again because of the same bug, it wasn't included. So it's good that it's now there.

I'm not sure if it makes sense to change the tests before there's a new release of ClojureScript with these changes.

**Comment by ikitommi** [comment]
Any idea when this would be merged in CLJS master? last stabile still seems to be `1.10.339`

**Comment by mfikes** [comment]
@ikitommi CLJS-2839 is actually in master. Assuming you are really asking about when the next ClojureScript release will be, last week David mentioned releasing soon: https://clojurians-log.clojureverse.org/cljs-dev/2018-09-13/1536880696.000100

**Comment by ikitommi** [comment]
Fixed in 0.8.1
